webbpsf/opds.py

```diff
@@ -1118,15 +1133,23 @@ def __init__(self, name='Unnamed OPD', opd=None, opd_index=0, transmission=None,
             Tuple giving V2,v3 coordinates as quantities, typically in arcminutes, or None to default to
             the master chief ray location between the two NIRCam modules.
         control_point_fieldpoint: str
-            Name of the field point where the OTE control point is located, on instrument defined by "control_point_instr". 
+            Name of the field point where the OTE control point is located, on instrument defined by "control_point_instr".
             Default: 'nrca3_full'.
             The OTE control point is the field point to which the OTE has been aligned and defines the field angles
             for the field-dependent SM pose aberrations.
+        npix : int
+            Size of OPD: npix x npix
 
         """
 
-        OPD.__init__(self, name=name, opd=opd, opd_index=opd_index, transmission=transmission, segment_mask_file=segment_mask_file)
+        OPD.__init__(self, name=name, opd=opd, opd_index=opd_index, transmission=transmission, segment_mask_file=segment_mask_file, npix=npix)
         self.v2v3 = v2v3
+        self.npix = npix
+
+        # Check that the shape of the OPD that has been passed, matches the npix parameters
+        if self.opd is not None:
+            if not self.opd.shape == (self.npix, self.npix):
+                raise ValueError(f"npix value of {self.npix} does not match shape of OPD provided: {self.opd.shape}")
```

Comment:
Optiona, but maybe this also moves up into the superclass init?  (That said, the other OPD linear model class doesn't get used at all any more and hasn't gotten used since IV&V analyses post JSC, so this class hierarchy is in some ways not as critical to maintain now, and it's equally effective to have this check here)